### DISCUSSION TITLE
Embedding into binary

### DISCUSSION THREAD

**Discussion by bencleary** [nats-io/nats.rs#752] [Q&A] [answered]
Is there a way to embed a NATS server into a rust application, i know this can be done with Go and just wondering if this was possible using Rust also?

**Comment by Jarema** [comment, marked as answer]
NATS server is written in Go, and you cannot embed it in Rust.
You can run the NATS server from within the Rust application like our tests do (that's the purpose `nats-server` crate), but that's just spawning a child process.

**Reply by steveej** [reply]
would calling the NATS Go embedding library from Rust via FFI work?

**Reply by Jarema** [reply]
No don't think so. You need full Go runtime to allow server to work. Not to mention the complexity of the FFI part.

**Reply by steveej** [reply]
an alternative to native embedding would be to listen to a Unix domain socket. is that supported today?
